File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* temp1 = head;
        int c = 0;
        
        if (n == 1 && temp1->next == nullptr)
            return nullptr;

        while (temp1 != nullptr) {
            c++;
            temp1 = temp1->next;
        }
        
        int t = c - n;
        temp1 = head;
        
        if (t == 0) {
            ListNode* newHead = head->next;
            delete head;
            return newHead;
        }
        
        while (t - 1 > 0) {
            temp1 = temp1->next;
            t--;
        }
        
        ListNode* temp2 = temp1->next;
        if (temp1->next != nullptr) {
            temp1->next = temp1->next->next;
        }
        delete temp2;

        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        temp=head
        c=0
        while temp!=None:
            temp=temp.next
            c+=1
        temp2=head
        k=c-n
        i=0
        if c==1:
            del head
            return
        if c==n and head:
           head=head.next
           return head

        while i<k-1:
            temp2=temp2.next
            i+=1
        if temp2.next:
           temp3=temp2.next
           temp2.next=temp3.next
        return head

        